examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp

```diff
@@ -493,12 +496,13 @@ CHIP_ERROR CastingServer::VerifyOrEstablishConnection(TargetVideoPlayerInfo & ta
         prevDeviceProxy->Disconnect();
     }
 
-    CastingServer::GetInstance()->mActiveTargetVideoPlayerInfo = targetVideoPlayerInfo;
+    CastingServer::GetInstance()->mActiveTargetVideoPlayerInfo = mCachedTargetVideoPlayerInfo[cacheIndex];
```

Comment:
Might be cleaner if you just maintain a pointer since you use this 3 times

```suggestion
    TargetVideoPlayerInfo *cachedTargetVideoPlayerInfo = &mCachedTargetVideoPlayerInfo[cacheIndex]
    CastingServer::GetInstance()->mActiveTargetVideoPlayerInfo = *cachedTargetVideoPlayerInfo;
    uint32_t delay = 0;
    if (cachedTargetVideoPlayerInfo->isAsleep()) {
       ...
```

Reply:
I feel like adding another \* cachedTargetVideoPlayerInfo pointer adds another level of indirection that makes the code harder to read. I'd much rather assign mActiveTargetVideoPlayerInfo the cached entry and then refer to the same mActiveTargetVideoPlayerInfo later on. I can switch to your suggestion if you want to push back though.

Reply:
It's a suggestion, i feel like having a separate pointer makes the code much easier to read since you're not parsing this super long string of `CastingServer::GetInstance()->mActiveTargetVideoPlayerInfo` - but practically speaking its really the same, i'll leave it up to you on whether or not you want to put it in.